libraries/tracking/streams/category.js

```diff
@@ -58,7 +58,7 @@ const categoryDataPreloaded$ = categoryIdChanged$
  * Emits when the category route comes active again after a legacy page was active.
  */
 const categoryRouteReappeared$ = pwaDidAppear$
-  .filter(({ pathname }) => pathname.startsWith(CATEGORY_PATH));
+  .filter(({ action }) => action.route.pattern.startsWith(CATEGORY_PATH));
```

Comment:
Please always check pattern with an equal condition: `...pattern === '${CATEGORY_PATH}/:categoryId'.`

Reply:
I had to update the `filter` step, since it now needs to compare against two different patterns, but it will correspond with your change request.